src/cpp/aiebu/src/common/utils.h

```diff
@@ -164,12 +164,26 @@ vector_of_string_to_vector_of_char(const std::vector<std::string>& args)
 }
 
 enum class fragment{
+  BEGIN_ANCHOR_RE,
+  END_ANCHOR_RE,
   HEX_RE,
   L_BRACK_RE,
-  R_BRACK_RE
+  R_BRACK_RE,
 };
 
 std::regex get_regex(const std::vector<fragment>& pattern);
 
+constexpr unsigned hexbase = 0x10;
+
+template <typename INTT> INTT to_int(const std::string &token) {
+  const unsigned long long result = (std::is_same<INTT, uint64_t>::value) ? std::stoull(token, nullptr, hexbase) :
+    std::stoul(token, nullptr, hexbase);
```

Comment:
This doesn't look very flexible.  It converts with base16, if that is really the intend, the function name should reflect this.  Else, I think the base should set to `0` to autodetect hexadecimal (0x), octal (0), or otherwise decimal.

The formatting of the template and function names makes it very hard to read. Better
```
template <typename IntType>
IntType 
to_int(const std::string& token)
{
  ...
}
```

Reply:
Done